pallets/runtime/src/bridge.rs

```diff
@@ -391,4 +423,39 @@ impl<T: Trait> Module<T> {
         }
         Ok(None)
     }
+
+    /// Handles a bridge transaction proposal immediately.
+    fn handle_bridge_tx_now(bridge_tx: BridgeTx<T::AccountId, T::Balance>) -> DispatchResult {
+        if let Some(PendingTx { did, bridge_tx }) = Self::issue(bridge_tx.clone())? {
+            <PendingTxs<T>>::mutate(did, |txs| txs.push(bridge_tx.clone()));
+            Self::deposit_event(RawEvent::Pending(PendingTx { did, bridge_tx }));
+        } else {
+            <HandledTxs<T>>::insert(&bridge_tx, true);
+            Self::deposit_event(RawEvent::Bridged(bridge_tx));
+        }
+        Ok(())
+    }
+
+    /// Handles a bridge transaction proposal after `timelock` blocks.
+    fn handle_bridge_tx_later(
+        bridge_tx: BridgeTx<T::AccountId, T::Balance>,
+        timelock: T::BlockNumber,
+    ) -> DispatchResult {
+        let current_block_number = <system::Module<T>>::block_number();
+        let unlock_block_number = current_block_number + timelock;
+        <TimelockedTxs<T>>::mutate(unlock_block_number, |txs| {
+            txs.push(bridge_tx);
+        });
+        Ok(())
+    }
+
+    /// Handles the timelocked transactions that are set to unlock at the given block number.
+    fn handle_timelocked_txs(block_number: T::BlockNumber) {
+        let txs = <TimelockedTxs<T>>::take(block_number);
```

Comment:
There may be a possibility that all those transactions can't be executed with in the same block because of the block gas limit and if that happens then those transactions will never be executed and always remain locked. We need some kind of manual dispatchable to execute those pending txn or some worker logic to execute those txn in batches.

Reply:
I'm afraid you may be right. This is even a bit worse. Instead of remaining timelocked, the transactions are currently removed with `take`. So we need to keep transactions that cannot be executed due to the gas limit for later reprocessing.

Reply:
@satyamakgec Can you check if I fixed this OK?

Reply:
Yeah @vkomenda changes make sense